src/heartbeats.cpp

```diff
@@ -0,0 +1,611 @@
+
+    // add the transfer data to a list of finished. To reduce the totals with its values
+    // when we consider progress is complete
+    mFinishedTransfers.push_back(std::move(pending));
+
+    mPendingTransfers.erase(transfer->getTag());
+
+    if (mPendingTransfers.empty())
+    {
+        clearFinshedTransfers(); // asume the sync is up-to-date: clear totals
+        assert(!mPendingUps && !mPendingDowns);
+        mTotalBytes = 0;
+        mTransferredBytes = 0;
+    }
+}
+
+void HeartBeatSyncInfo::clearFinshedTransfers()
+{
+    for (const auto &pending : mFinishedTransfers)
+    {
+        // reduce globals by the last known data
+        mTotalBytes -= pending->totalBytes;
+        mTransferredBytes -= pending->transferredBytes;
+    }
+
+    mFinishedTransfers.clear();
+}
+
+void HeartBeatSyncInfo::setHeartBeatId(const handle &heartBeatId)
+{
+    mHeartBeatId = heartBeatId;
+}
+
+int HeartBeatSyncInfo::syncTag() const
+{
+    return mSyncTag;
+}
+
+void HeartBeatSyncInfo::setTransferredBytes(long long value)
+{
+    if (mTransferredBytes != value)
+    {
+        mTransferredBytes = value;
+        updateLastActionTime();
+    }
+}
+
+void HeartBeatSyncInfo::setTotalBytes(long long value)
+{
+    if (mTotalBytes != value)
+    {
+        mTotalBytes = value;
+        updateLastActionTime();
+    }
+}
+
+void HeartBeatSyncInfo::setLastSyncedItem(const mega::MegaHandle &lastSyncedItem)
+{
+    if (mLastSyncedItem != lastSyncedItem)
+    {
+        mLastSyncedItem = lastSyncedItem;
+        updateLastActionTime();
+    }
+}
+
+void HeartBeatSyncInfo::setPendingDowns(uint32_t pendingDowns)
+{
+    if (mPendingDowns != pendingDowns)
+    {
+        mPendingDowns = pendingDowns;
+        updateLastActionTime();
+    }
+}
+
+void HeartBeatSyncInfo::setPendingUps(uint32_t pendingUps)
+{
+    if (mPendingUps != pendingUps)
+    {
+        mPendingUps = pendingUps;
+        updateLastActionTime();
+    }
+}
+
+void HeartBeatSyncInfo::setStatus(const Status &status)
+{
+    if (mStatus != status)
+    {
+        mStatus = status;
+        updateLastActionTime();
+    }
+}
+
+void HeartBeatSyncInfo::setLastAction(const m_time_t &lastAction)
+{
+    mLastAction = lastAction;
+}
+
+void HeartBeatSyncInfo::updateLastActionTime()
+{
+    setLastAction(m_time(nullptr));
+}
+
+void HeartBeatSyncInfo::setLastBeat(const m_time_t &lastBeat)
+{
+    mLastBeat = lastBeat;
+}
+
+m_time_t HeartBeatSyncInfo::lastBeat() const
+{
+    return mLastBeat;
+}
+
+void HeartBeatSyncInfo::onCommandToBeDeleted(Command *command)
+{
+    if (mRunningCommand == command)
+    {
+        mRunningCommand = nullptr;
+    }
+}
+
+MegaHeartBeatMonitor::MegaHeartBeatMonitor(MegaClient *client)
+    : mClient(client)
+{
+}
+
+MegaHeartBeatMonitor::~MegaHeartBeatMonitor()
+{
+}
+
+void MegaHeartBeatMonitor::reset()
+{
+    mPendingBackupPuts.clear();
+    mHeartBeatedSyncs.clear();
+    mPendingSyncUpdates.clear();
+    mLastBeat = 0;
+}
```

Comment:
This method seems to be unused in the code (and incomplete, since it does not reset all the class members).
Should we simply remove it?

Reply:
Solved!